pkg/manager/manager.go

```diff
@@ -217,3 +203,30 @@ func (m *mcManager) Engage(ctx context.Context, name string, cl cluster.Cluster)
 	}
 	return nil //nolint:govet // cancel is called in the error case only.
 }
+
+func (m *mcManager) GetManager(ctx context.Context, clusterName string) (manager.Manager, error) {
+	cl, err := m.GetCluster(ctx, clusterName)
+	if err != nil {
+		return nil, err
+	}
+	return &scopedManager{
+		Manager: m,
+		Cluster: cl,
+	}, nil
+}
+
+var _ manager.Manager = &scopedManager{}
+
+type scopedManager struct {
+	Manager
```

Comment:
Is this a `mcmanager.Manager` on purpose or should it be a `manager.Manager`? Doesn't this allow for confusing things like:

```go
scopedManager, err := m.GetManager(ctx, "cluster-a")
[...]
scopedScopedManager, err := scopedManager.GetManager(ctx, "cluster-b")
```

Reply:
`GetManager` returns a `manager.Manager`, not `scopedManager`.

Reply:
or in other words:
- `mcmanager.Manager == manager.Manager - cluster.Cluster`
- => `mcmanager.Manager + cluster.Cluster == manager.Manager`

Reply:
Ah, I see. My bad. So while the `scopedManager` struct has this func from embedding the manager, it's hidden by the interface.